utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java

```diff
@@ -30,12 +30,14 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
 import software.amazon.awssdk.utils.StringUtils;
 import software.amazon.awssdk.utils.Validate;
 
 /**
  * A set of utilities that assist with HTTP message-related interactions.
  */
+@SdkProtectedApi
```

Comment:
Public? Aren't there cases where customers need to url encode?

Reply:
Before GA, sure. It's not ready to be public right now, though.